### PR TITLE
Spago module

### PR DESCRIPTION
**Pull request by SeungheonOh**
### Summary of changes

...

### Tested on
- [ ] [agora](https://github.com/Liqwid-Labs/agora)
- [ ] [liqwid-libs](https://github.com/Liqwid-Labs/liqwid-libs)
- [ ] [oracle-offchain](https://github.com/Liqwid-Labs/oracle-offchain)
- [ ] ...

### CODE REVIEW

File: nix/offchain.nix [hunk 1]
```diff
@@ -206,13 +218,21 @@ in
...
+
                   Added in: 2.3.0.
                 '';
                 default = null;
                 type = types.nullOr (types.raw or types.unspecified);
               };
 
+              spagoOverride = lib.mkOption {
+                description = ''
+                  Override `packages.dhall` and use given flake for spago module.
```
Comment (emiflake) [inline nix/offchain.nix:230]:
```suggestion
                  Override `packages.dhall` and use given flake for spago module. 
                  
                  Added in: 2.3.1.
```

File: nix/offchain.nix [hunk 2]
```diff
@@ -190,6 +190,18 @@ in
             };
           };
 
+          # Inorder to use as a spago module, it most be a store path to the
+          # spago module repository, and must have git revison that is for
+          # ensuring version.
+          spagoModule = lib.mkOptionType {
+            name = "spagoModule";
+            description = "noun";
```
Comment (emiflake) [inline nix/offchain.nix:198]: Needs a description

**Review by emiflake — CHANGES_REQUESTED** [review]
Looks good! Just a couple changes.

**Comment by emiflake** [comment]
Additionally, please add to the CHANGELOG.

File: nix/offchain.nix (hunk 2 above)
Comment (SeungheonOh) [reply nix/offchain.nix:198]: https://github.com/NixOS/nixpkgs/blob/7b2a482ea0240ef989001c6a6c28b47db7790483/lib/types.nix#L113

`noun` is a proper description

Comment (emiflake) [reply nix/offchain.nix:198]: Ah, well I think this is supposed to _describe_ what the option does, as opposed to be a specific tag. Perhaps we haven't followed the guidelines properly for this, but if you look around, you'll see that we use a specific format for descriptions, and they all tag using `Added in:`. This description actually doesn't ever get used anywhere. It's basically just documentation atm.

Comment (SeungheonOh) [reply nix/offchain.nix:198]: Ah, okay. Thanks

**Review by emiflake — APPROVED** [review]

**Pull request merged**